File: src/leetcode/0119_pascals_triangle_2.rs
fn main() {
    println!("{:?}", get_row(0));
    println!("{:?}", get_row(1));
    println!("{:?}", get_row(2));
    println!("{:?}", get_row(3));
    println!("{:?}", get_row(4));
    println!("{:?}", get_row(5));
}

pub fn get_row(row_index: i32) -> Vec<i32> {
    if row_index == 0 { return vec![1]; }
    let mut previous: Vec<i32> = vec![1,1];

    for i in 1..=row_index {
        let mut new: Vec<i32> = vec![1];
        for j in 1..i {
            new.push(previous[(j-1) as usize] + previous[j as usize]);
        }
        new.push(1);
        previous = new;
    }
    previous
}

File: src/leetcode/0119_pascals_triangle_2.py
class Solution:
    def getRow(self, rowIndex: int) -> list[int]:
        row = [1]
        for i in range(1, rowIndex + 1):
            new_row = [1]
            for j in range(1, i):
                new_row.append(row[j - 1] + row[j])
            new_row.append(1)
            row = new_row
        return row


s = Solution()
print(s.getRow(3))
print(s.getRow(0))
print(s.getRow(1))
print(s.getRow(4))
